### PR TITLE
Variables: Add variable query editor

### PR DESCRIPTION
**Pull request by lean257**
Related to #113 
Screen Shot 2022-05-02 at 6 34 28 PM

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export function frameToMetricFindValues(df: DataFrame): MetricFindValue[] {
+    const res:MetricFindValue[] = [];
+    const view = new DataFrameView<SimpleResult>(df);
+    view.forEach(item => {
+        const id = item.id;
+        const name = item.name;
+        if (id && name) {
+            res.push({text: name, value:id});
+        }
+        const value = id || name;
```
Comment (lean257) [inline src/utils.ts:17]: should this be inside an else so that we only push either or, instead of both scenarios?

Comment (ryantxu) [reply src/utils.ts:17]: it is weird -- in 7.5+, the DataFrameView would let us check fields outside the loop, but I don't think we want to bump the requirements for that.

I think this is OK as a first pass

File: src/components/query/QueryEditor.tsx [hunk 2]
```diff
@@ -19,7 +19,12 @@ export const firstLabelWith = 15;
 
 export class QueryEditor extends PureComponent<Props> {
   onQueryTypeChange = (sel: SelectableValue<QueryType>) => {
-    const { onChange, query, onRunQuery } = this.props;
+    const { onChange, query } = this.props;
+    // hack to use QueryEditor as VariableQueryEditor
+    let { onRunQuery } = this.props;
+    if (!onRunQuery) {
+      onRunQuery = () => {};
+    }
```
Comment (ryantxu) [inline src/components/query/QueryEditor.tsx:27]:
```suggestion
    const onRunQuery = this.props.onRunQuery ?? () => {};
```
maybe a little better 🤷

File: src/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
+import { dataFrameFromJSON, DataFrameJSON } from '@grafana/data';
+
+import { frameToMetricFindValues } from './utils';
+
+describe('Test utils', () => {
+  it('convert simple values', () => {
+    const df = dataFrameFromJSON(listAssetsResponse as DataFrameJSON);
+    expect(frameToMetricFindValues(df)).toMatchInlineSnapshot(`
```
Comment (ryantxu) [inline src/utils.test.ts:8]: hymm -- something funky here.  The result has 1 row, so I would expect 1 result 🤔

File: src/utils.ts (hunk 1 above)
Comment (ryantxu) [reply src/utils.ts:17]: dooh -- yes obviously!!! 🤦 

good thing we are writing tests!!!

**Review by ryantxu — APPROVED** [review]
LGTM

**Pull request merged**